"Possible null reference return" in auto-generated compiled model

When running `dotnet ef dbcontext optimize -c Database` with version `6.0.0-rc.2.21421.3` of the `dotnet-ef` tool, the generated code contains a compiler warning in `DatabaseModel.cs`:

```c#
private static DatabaseModel? _instance;
public static IModel Instance => _instance; // <-- warning CS8603: Possible null reference return
```

Not a big issue, but thought it might be helpful to report.

--- a/src/EFCore.Design/Scaffolding/Internal/CSharpRuntimeModelCodeGenerator.cs
+++ b/src/EFCore.Design/Scaffolding/Internal/CSharpRuntimeModelCodeGenerator.cs
@@ -176,7 +176,7 @@ namespace Microsoft.EntityFrameworkCore.Scaffolding.Internal
     _instance = model;
 }")
                     .AppendLine()
-                    .Append("private static ").Append(className).AppendLine(nullable ? "? _instance;" : " _instance;")
+                    .Append("private static ").Append(className).AppendLine(" _instance;")
                     .AppendLine("public static IModel Instance => _instance;")
                     .AppendLine()
                     .AppendLine("partial void Initialize();")

--- a/test/EFCore.Design.Tests/Scaffolding/Internal/CSharpRuntimeModelCodeGeneratorTest.cs
+++ b/test/EFCore.Design.Tests/Scaffolding/Internal/CSharpRuntimeModelCodeGeneratorTest.cs
@@ -876,7 +876,7 @@ namespace TestNamespace
             _instance = model;
         }
 
-        private static BigContextModel? _instance;
+        private static BigContextModel _instance;
         public static IModel Instance => _instance;
 
         partial void Initialize();
@@ -3619,7 +3619,8 @@ namespace TestNamespace
                     BuildReference.ByName("Newtonsoft.Json"),
                     BuildReference.ByName(typeof(CSharpRuntimeModelCodeGeneratorTest).Assembly.GetName().Name)
                 },
-                Sources = scaffoldedFiles.ToDictionary(f => f.Path, f => f.Code)
+                Sources = scaffoldedFiles.ToDictionary(f => f.Path, f => f.Code),
+                NullableReferenceTypes = options.UseNullableReferenceTypes
             };
 
             var assembly = build.BuildInMemory();

--- a/test/EFCore.Design.Tests/TestUtilities/BuildSource.cs
+++ b/test/EFCore.Design.Tests/TestUtilities/BuildSource.cs
@@ -109,8 +109,7 @@ namespace Microsoft.EntityFrameworkCore.TestUtilities
                     OutputKind.DynamicallyLinkedLibrary,
                     nullableContextOptions: NullableReferenceTypes ? NullableContextOptions.Enable : NullableContextOptions.Disable,
                     reportSuppressedDiagnostics: false,
-                    // TODO: Uncomment when #18950 is fixed
-                    //generalDiagnosticOption: ReportDiagnostic.Error,
+                    generalDiagnosticOption: ReportDiagnostic.Error,
                     specificDiagnosticOptions: new Dictionary<string, ReportDiagnostic>()
                     {
                         { "CS1030", ReportDiagnostic.Suppress },

